File: superheroes/superhero-08-01/src/model/gemini.js
// https://github.com/google-gemini/generative-ai-js
import { GoogleGenerativeAI } from "@google/generative-ai";
import 'dotenv/config'

export class GeminiModel {
  constructor() {
    const apiKey = "AIzaSyDr_lWrDot0YEsCy3bL_9laXKbqouQYcNA";
    const genAI = new GoogleGenerativeAI(apiKey);

    this.model = genAI.getGenerativeModel({ model: "gemini-1.5-flash-latest" });
  }

  async process(prompt) {
    const result = await this.model.generateContent([prompt]);

    return result.response.text();
  }
}

File: superheroes/superhero-08-02/backend/api/services/gemini.py
import os
import google.generativeai as genai

class GeminiAPI:

    def __init__(self):
        genai.configure(api_key = os.getenv('GEMINI_API_KEY'))

    def prompt_gemini(self, prompt):
        model = genai.GenerativeModel("gemini-1.5-flash")
        response = model.generate_content(prompt)
        return response.text
    
geminiAPI = GeminiAPI()
